frontend/app.js

```diff
@@ -5,19 +5,23 @@ const express = require('express');
 const path = require('path');
 const app = express();
 
-/* This code makes sure that every request that matches a static file in the build folder, it serves that file. */
+/* This code makes sure that every request that matches
+a static file in the build folder, it serves that file. */
 app.use(express.static(path.join(__dirname, 'build')));
 
 
-/* This code makes sure that any request that does not matches a static file in the build folder, will just serve index.html. Client side routing is going to make sure that the correct content will be loaded from there. */
-app.get('/*', function (req, res) {
-    res.sendFile(path.join(__dirname, 'build', 'index.html'));
-});
+/* This code makes sure that any request that does not matches a static
+file in the build folder, will just serve index.html.
+Client side routing is going to make sure that the correct content will
+be loaded from there. Commented when developing so proxy can work */
+// app.get('/*', function (req, res) {
+//     res.sendFile(path.join(__dirname, 'build', 'index.html'));
+// });
```

Comment:
Do we still need this?

Reply:
It is commented so that proxy works in development. But once deployed it would be good to have.

Reply:
Generally, we'd want this to be conditionally executed only if run in a certain mode (e.g. do not execute if running on local development server). That's definitely out of scope for this PR, and might be too difficult to be worth it anyways, but might be good to look into.

It looks like you can detect whether or not it's a local dev server pretty easily in Java code, and then that could be injected into the index.html page that gets written to the client:
https://cloud.google.com/appengine/docs/standard/java/tools/using-local-server#detecting_the_application_runtime_environment

Reply:
Thanks! I look into it.